server/apps/preferences.py

```diff
@@ -154,13 +154,12 @@ def on_update(self, updates, original):
     def update_session_prefs(self, updates, existing_session_preferences, session_id):
         session_prefs = updates.get(_session_preferences_key)
         if session_prefs is not None:
-            for k in ((k for k, v in session_prefs.items() if k not in superdesk.default_session_preferences)):
+            for k in (k for k, v in session_prefs.items() if k not in superdesk.default_session_preferences):
                 raise ValidationError('Invalid preference: %s' % k)
 
             existing = existing_session_preferences.get(session_id, {})
             existing.update(session_prefs)
-            existing_session_preferences[session_id] = existing
```

Comment:
thing is that [in saveLocally](https://github.com/superdesk/superdesk/blob/master/client/app/scripts/superdesk/services/preferencesService.js#L33) it doesn't know about this structure so those settings changes are not available until you reload the page